packages/dashboard/src/components/widgets/primitives/input/index.tsx

```diff
@@ -1,22 +1,32 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { NonCancelableEventHandler } from '@cloudscape-design/components/internal/events';
 import { Button, Select, SelectProps } from '@cloudscape-design/components';
 import SpaceBetween from '@cloudscape-design/components/space-between';
 import { InputWidget as InputWidgetType } from '../../../../types';
 
 export type InputWidgetProps = InputWidgetType & { readOnly: boolean };
 
 const Input: React.FC<InputWidgetProps> = ({ readOnly, ...widget }) => {
-  const [selectedOption, setSelectedOption] = useState<SelectProps.Option>(widget.options[0]);
+  const [selectedOption, setSelectedOption] = useState<SelectProps.Option | null>(widget.options[0]);
+  const options = widget.options.map(({ label }) => ({ label, value: label }));
+  const disabled = !readOnly || options.length === 0;
```

Comment:
when will it be readonly?

Reply:
I'm guessing this was to appease eslint accessibility warnings?

Reply:
It will be read only when the dashboard is in view mode (no side panels and grid is locked)

Reply:
No, nothing to do with eslint. Talked to UX and we do not want to widget to be used in edit mode.

Reply:
input needs to be usuable in 'view mode', operators need to be able add these notes, even if they don't have access to edit the dashboard.

Reply:
Exactly, that is why this code will allow the input widget to be usable in "view mode" and only prevent users from using it in "edit mode"